src/Core/Resolvers/SqlPaginationUtil.cs

```diff
@@ -339,8 +342,8 @@ RuntimeConfigProvider runtimeConfigProvider
                 {
                     string columnName = GetExposedColumnName(entityName, column.ColumnName, sqlMetadataProvider);
 
-                    if (!afterDict.ContainsKey(columnName) ||
-                        afterDict[columnName].Direction != column.Direction)
+                    if (!exposedFieldNameToBackingColumn.ContainsKey(columnName) ||
```

Comment:
usage of var name `columnName` could be revised to be `exposedFieldName` to directly align with the dictionary `exposedFieldNameToBackingColumn`. To be consistent with other code in this file, you'd want to stick with 'fieldName' usage for user facing field names, and 'columnName' for any metadata directly mapping to the database/not exposed to the user.